### PR TITLE
FS-1622 Prevent auto-creation of applications on account creation

### PR DESCRIPTION
**Pull request by harryyo**
### Prevent auto-creation of applications on account creation

- Add CREATE_APPLICATION_ON_ACCOUNT_CREATION config flag defaulted to False
- Use as a toggle to enable/disable auto-creation of applications on account creation

### CODE REVIEW

File: models/account.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def get_magic_link(
                 ],  # noqa
                 NotifyConstants.MAGIC_LINK_FUND_NAME_FIELD: fund.name,
             }
-            if fund_id and round_id and new_account:
+            if fund_id and round_id and new_account and Config.CREATE_APPLICATION_ON_ACCOUNT_CREATION:
```
Comment (srh-sloan) [inline models/account.py:127]: Should we add some unit tests around this? That might have highlighted the change in behaviour when adding a default value.

**Review by Adam-W1 — COMMENTED** [review]
Amazing the difference in readability a flag can make 👏  unit tests would be nice :_)

File: models/account.py (hunk 1 above)
Comment (harryyo) [reply models/account.py:127]: Yes! Added

**Review by srh-sloan — APPROVED** [review]
Great! 👍

**Review by Adam-W1 — APPROVED** [review]
LGTM ⚡

**Pull request merged**